chore: Run tests on default ClickHouse 22.8 (LTS)

macobo opened the pull request:
This PR:
- Updates us to support ClickHouse 22.8
- Updates tests to run against ClickHouse 22.8
- Drops support for ClickHouse versions prior to 22.3

Merge this when rolling out ClickHouse 22.8 on EU/US Cloud - without it tests will fail.

Related: https://github.com/PostHog/posthog/issues/12902, https://github.com/PostHog/posthog/issues/12967

macobo committed:
chore: Test running using ClickHouse 22.8 (LTS)

Related: https://github.com/PostHog/posthog/issues/12902

macobo committed:
Make optimize_move_to_prewhere version dependant

macobo committed:
Run E2E tests using CH 22.8 in this branch

macobo committed:
Run plugin-server tests

macobo committed:
Run tests against 22.8 by default

macobo committed:
Merge remote-tracking branch 'origin/master' into ch-228-testing

macobo committed:
Drop support for clickhouse versions below 22.3

ellie commented:
would it be possible to get this rebased and merged please?

ellie commented:
I've tried to rebase this myself but there has been a bunch of CI changes around clickhouse and async migrations

cc @Twixes and @macobo because I don't have the context there

macobo committed:
Merge remote-tracking branch 'origin/master' into ch-228-testing

macobo commented:
Updated the PR to run tests on 22.8.

On previous changeset:
- We already added support for running 22.8 in https://github.com/PostHog/posthog/pull/13330
- We are not prioritizing dropping support of versions prior to 22.3 (something for infra backlog)
- Most of the CI problems were already resolved

Let's merge once EU and US clouds are on 22.8
